share/blob_tx.go

```diff
@@ -0,0 +1,78 @@
+package share
+
+import (
+	"errors"
+
+	v1 "github.com/celestiaorg/go-square/proto/blob/v1"
+	"google.golang.org/protobuf/proto"
+)
+
+const (
+	// ProtoBlobTxTypeID is included in each encoded BlobTx to help prevent
+	// decoding binaries that are not actually BlobTxs.
+	ProtoBlobTxTypeID = "BLOB"
+)
+
+type BlobTx struct {
```

Comment:
Is the idea of making go-square hold actual transactions? Isn't that a concern of the app?